### PR TITLE
Add Export button missing on new Product Page V2

### PR DESCRIPTION
**Pull request by OrionCodeDev**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | This PR introduces the much-requested "Export" button to the new Product Page V2 in PrestaShop, addressing a significant gap in the platform's functionality. Previously, users experienced a workflow interruption due to the absence of an export feature directly from the product list, which is essential for exporting data to CSV or Excel formats for various operational needs. This update brings consistency across the platform's interfaces, significantly enhances operational efficiency, enables seamless data integration with external systems, supports advanced reporting and analytics, and meets user expectations for feature parity with previous versions. By reinstating the export feature, this PR ensures that PrestaShop remains a comprehensive tool for e-commerce businesses, catering to their evolving needs and maintaining its utility and user satisfaction.
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| How to test?      | Turn new product page -> Go to the product list -> click config icon -> click "Export"
| Fixed issue or discussion?     | #35485
| Sponsor company   | Code Olympus

### CODE REVIEW

**Comment by OrionCodeDev** [comment]
I must do something else or ping somebody for show ready state of this pull? Sorry, but it's my first pull request.

**Comment by boherm** [comment]
> I must do something else or ping somebody for show ready state of this pull? Sorry, but it's my first pull request.

Hi @ChronoBrake! Thanks for your first contribution! ;)
No, we just need to review it and test it before merging.

Some steps could take some times, but we will check your PR as soon as possible!

File: src/PrestaShopBundle/Resources/config/routing/admin/sell/catalog/products_v2/product.yml [hunk 1]
```diff
@@ -34,6 +34,14 @@ admin_products_preview:
   requirements:
     productId: \d+
 
+admin_products_export:
+  path: /export
+  methods: [ GET ]
+  defaults:
+    _controller: PrestaShopBundle\Controller\Admin\Sell\Catalog\Product\ProductController::exportAction
+    _legacy_controller: AdminProducts
+    _legacy_link: AdminProducts:exportproduct
```
Comment (boherm) [inline src/PrestaShopBundle/Resources/config/routing/admin/sell/catalog/products_v2/product.yml]: The `exportproduct` doesn't seems to be in the legacy `controllers/admin/AdminProductsController.php`.
Maybe, we can simply do this:

```suggestion
    _legacy_link: AdminProducts
```

**Review by boherm — CHANGES_REQUESTED** [review]
LGTM, thanks @ChronoBrake!
Just a little comment about the `_legacy_link` ;)

File: src/PrestaShopBundle/Resources/config/routing/admin/sell/catalog/products_v2/product.yml (hunk 1 above)
Comment (OrionCodeDev) [reply src/PrestaShopBundle/Resources/config/routing/admin/sell/catalog/products_v2/product.yml]: @boherm yeah, it's will be better. Commited

**Comment by OrionCodeDev** [comment]
Added @boherm suggestion.

**Review by boherm — APPROVED** [review]

**Comment by OrionCodeDev** [comment]
@boherm i must do anything else or it's need time for devs to verify?

**Comment by boherm** [comment]
Hi @ChronoBrake, yes for now we just need some time on our side to review your PR.
I've start the CI, to make a little check about coding standard on your work ;)

**Comment by OrionCodeDev** [comment]
Okay, so i will be here if you wanna ask for repair or changing something. All checks green, so i think it's good. @boherm

**Review by nicosomb — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @ChronoBrake ,

Auto test ✅ : https://github.com/florine2623/testing_pr/actions/runs/9369588578

Export is working as expected.
Tested with all products.
Tested with filtered list of products (one and multiple filters).
Tested with no products.

It is QA ✅ 
Thanks!

**Comment by PascalB67** [comment]
Hello it's ok for this version (8.17), 
In the middle of developing a tool processing "data export", ... my heart rate increased significantly ... :)
The export limit seems to be based on the display of the page... ORDER BY id_product asc LIMIT 100. (In parentheses it is no longer possible to display more than 100 products per page... oops, bad idea). 
The best is to use a module to export, this is more professional for the end user
Thanks

**Pull request merged**